src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2369,7 +2419,7 @@ private BoundExpression BindIncrementOperator(CSharpSyntaxNode node, ExpressionS
                 resultConversion,
                 resultKind,
                 originalUserDefinedOperators,
-                operandType,
+                forceDynamicResult ? Compilation.DynamicType : operandType,
```

Comment:
>forceDynamicResult ? Compilation.DynamicType : operandType, [](http://example.com/codeflow?start=16&length=59)

Here's we'd use the helper method for return type, in response to use of `AdjustAssignmentTarget` above. Same comment applies at other locations where `AdjustAssignmentTarget` is used